docs/encyclopedia/application-message-passing.mdx

```diff
@@ -342,16 +342,21 @@ The following content applies in every SDK except the Go SDK. See below.
 #### Exceptions in Signals
 
 In Signal handlers, throw [Application Failures](/references/failures#application-failure) only for unrecoverable errors, because the entire Workflow will fail.
+Similarly, allowing a failing Activity or Child Workflow to exhaust its retries, so that it throws an [Activity Failure](https://docs.temporal.io/references/failures#activity-failure) or [Child Workflow Failure](https://docs.temporal.io/references/failures#child-workflow-failure) will cause the entire Workflow to fail.
+Note that for Activities, this will only happen if you change the default Activity [Retry Policy](https://docs.temporal.io/encyclopedia/retry-policies), since by default they retry forever.
 If you throw any other exception, by default, it will cause a [Workflow Task Failure](/references/failures#workflow-task-failures). This means the Workflow will get stuck and will retry the handler periodically until the exception is fixed, for example by a code change.
 
 #### Exceptions in Updates
 
-When you want to fail the Update and for the client to receive that error:
+Doing any of the following will fail the Update and cause the client to receive the error:
 
-- Reject the Update by throwing any exception from your Validator, OR
-- Throw an [Application Failure](/references/failures#application-failure) from your Update handler. Unlike with Signals, the Workflow will keep going.
+- Reject the Update by throwing any exception from your [Validator](https://docs.temporal.io/encyclopedia/workflow-message-passing#update-validators).
```

Comment:
> throwing

Note, most of our SDKs don't have the concept of "throw"

> exceptions

Note, some languages call these errors.

Not sure if anything needs to change, but to a reader this can appear very Java/.NET centric.

Reply:
That's intentional. "Throw" is the term used in Java, Typescript, and .NET. Python and Ruby use "raise" which is idiosyncratic and in the minority; they'll understand "throw". Go has a separate section below due to the difficulty of describing error handling in one sentence in a way that applies across Go and exception-based languages.